listen.js

```diff
@@ -1,30 +1,24 @@
-const pull = require('pull-stream')
+const flumeView = require('flumeview-reduce')
 const { isValid } = require('./spec/group/add-member')
 
 module.exports = {
   addMember
 }
 
-function addMember (ssb) {
-  var listeners = []
+const VERSION = 1
+function addMember (ssb, emit) {
+  // HACK: leveraging flume to access stream of newest messages
+  ssb._flumeUse('add-member-dummy-index', flumeView(
+    VERSION,
+    (_, msg) => {
+      if (msg.value.author === ssb.id) return _ // ignore messages I write
+      if (!isValid(msg)) return _
 
-  function subscribe (listener) {
-    listeners.push(listener)
-  }
-
-  // TODO problem is this doesn't track what group/add-member have been processed
-  // and watching "live" doesn't really guarentee ...
-  // I think we need a flumeview here because that tracks what's been past it / processed.
-
-  pull(
-    ssb.messagesByType({ type: 'group/add-member', private: true, live: true }),
-    pull.filter(m => m.sync !== true), // live queries emit { sync: true } when up to speed!
```

Comment:
one thing we could do with using `messagesByType` is count count how many new rebuilds should be called after we are added to a new group ... and wait till the indexing is all done before calling another rebuild. The problem with the new code (and existing code?) is it might immediately call a new rebuild each time it discovers a new add-member message in the new group... and so start again _many many times_ instead of just twice say ...